mne_bids/tsv_handler.py

```diff
@@ -0,0 +1,147 @@
+    output = ''
+    n_rows = len(data[list(data.keys())[0]])
+    # write headings
+    output += '\t'.join(list(data.keys())) + '\n'
+
+    # write column data
+    for idx in range(n_rows):
+        row_data = list(str(data[key][idx]) for key in data)
+        output += '\t'.join(row_data)
+        if idx != n_rows - 1:
+            output += '\n'
+
+    with open(fname, 'w') as f:
+        f.write(output)
+
+
+def _combine(data1, data2, drop_column=None):
+    """Add two OrderedDict's together with optional dropping of repeated data.
+
+    Parameters
+    ----------
+    data1 : collections.OrderedDict
+        Original OrderedDict.
+    data2 : collections.OrderedDict
+        New OrderedDict to be added to the original.
+    drop_column : str
+        Name of the column to check for duplicate values in.
+        Any duplicates found will be dropped from the original data array (ie.
+        most recent value are kept).
+
+    Returns
+    -------
+    data : collections.OrderedDict
+        The new combined data.
+    """
+    new_data = data1.copy()
+    for key, value in data2.items():
+        new_data[key].extend(value)
+    if drop_column in new_data:
+        column_data = new_data[drop_column]
+        values_count = Counter(column_data)
+        for key, value in values_count.items():
+            # find the locations of the first n-1 values and remove the rows
+            for _ in range(value - 1):
+                idx = new_data[drop_column].index(key)
+                for column in new_data.values():
+                    del column[idx]
+    return new_data
+
+
+def _drop(data, values, column):
+    """Remove rows from the OrderedDict.
+
+    Parameters
+    ----------
+    data : collections.OrderedDict
+        Data to drop values from.
+    values : list
+        List of values to drop. Any row containing this value in the specified
+        column will be dropped.
+    column : string
+        Name of the column to check for the existence of `value` in.
+
+    Returns
+    -------
+    data : collections.OrderedDict
+        Copy of the original data with 0 or more rows dropped.
+    """
+    new_data = data.copy()
+    mask = np.in1d(new_data[column], values, invert=True)
+    for key in new_data.keys():
+        new_data[key] = np.array(new_data[key])[mask].tolist()
+    return new_data
+
+
+def _contains_row(data, row_data):
+    """Whether the specified row data exists in the OrderedDict.
+
+    Parameters
+    ----------
+    data : collections.OrderedDict
+        OrderedDict to check.
+    row_data : list
+        List of values to be searched for. This must match the contents of a
+        row exactly for a positive match to be returned.
+    """
+    mask = None
+    for idx, column_data in enumerate(data.values()):
+        column_mask = np.in1d(np.array(column_data), row_data[idx])
+        mask = column_mask if mask is None else (mask & column_mask)
+    return np.any(mask)
+
+
+def _prettyprint(data, rows=5):
```

Comment:
I would call it `_tsv_to_str` or something like that. There is not printing business here